File: leetcode_go/greedy_algorithms/gfg_minimum_number_of_coins.go
/**
 * GFG: Minimum Number Of Coins
 * Link: https://www.geeksforgeeks.org/problems/-minimum-number-of-coins4426/1
*/

package main
import(
	"fmt"
)

func findMin(n int)  int {
	coins := []int{10,5,2,1}
	var total_coins int = 0
	for _, coin := range coins {
		q := n/coin;
		if q!=0 {
			total_coins += q
			n = n - q*coin
		}
	}
	return total_coins
}

func main(){
	fmt.Println(findMin(121))
}

File: leetcode_rs/greedy_algorithms/gfg_minimum_number_of_coins.rs
/**
 * GFG: Minimum Number Of Coins
 * Link: https://www.geeksforgeeks.org/problems/-minimum-number-of-coins4426/1
*/

fn find_min(mut n: i32) -> i32{
    let coins: Vec<i32> = vec![10,5,2,1];
    let mut total_coins: i32 = 0;
    for coin in coins.iter() {
        let q: i32 = n/coin;
        if q!=0 {
            total_coins += q;
            n = n - q*coin;
        }
    }
    total_coins
}

fn main(){
    println!("{}", find_min(121));
}